### PR TITLE
add js to handle multiple pressed keys

### PR DESCRIPTION
**Pull request by SepsiLaszlo**
## What is new?

- Add basic logic for detecting if multiple keys are pressed down at the same time.
- Search bar get focused on `ctrl + /`
- PointDetailComments are submitted on `ctrl + enter`

## Implementation details
- I declared a global hash that contains `true` for every key that is currenty are down
- This values are udated on every key up and down
- The function that listens for the keydown event to update the has, must be run before the value is read from the has in order to be accurate
  - To provide this I made sure that this function are subscribed first to `keydown`, because the callbacks happen in this order
  - This implementation can be prone to some errors that are hard to debug, if the subscription order is changed
  - I considered implementing a more roboust solution with a separate Publisher-Subscriber pattern but it made the code much less readable and less usable

Any thoughts on this?
I think this solution could work well for our use case. Maybe we should add some comment to warn future developers to the dangers.
If there is a more reliable way to implement this that does not intoduce too muh complexity let me know.

### CODE REVIEW

**Comment by balintkiraly** [comment]
Instead of the global hash you can easily use a simple listener with the check, so something like this:
```
document.addEventListener("keydown", e => {
  if (e.ctrlKey && e.key === '/') {
      const element = document.getElementById('search-field');
      element.focus();
  }
});
```
In this case we don't need to care the order of the listener and I think your solution won't work in some cases even if we pay attention to the order because we can be sure the listener will run in order.

By the way, do we really want to use the Ctrl+/ shortcut for the search? The simple `/` sounds better and more common as I see it.

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <%= render 'favicon' %>
   <%= stylesheet_link_tag    'application', media: 'all' %>
+  <%= javascript_include_tag 'key_press_listener' %>
```
Comment (balintkiraly) [inline app/views/layouts/application.html.erb:10]: It should be required in the application.js so we don't need to include it here.
You can move it to the `application` folder 
> //= require_tree ./application/

or you can explicitly import in the `application.js` file

File: config/initializers/assets.rb [hunk 2]
```diff
@@ -16,4 +16,5 @@
 # Rails.application.config.assets.precompile += %w( admin.js admin.css )
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 # Rails.application.config.assets.precompile += %w( search.js )
+Rails.application.config.assets.precompile += %w( key_press_listener.js )
```
Comment (balintkiraly) [inline config/initializers/assets.rb:19]: If it's in the application.js, so we don't need add this line
> #application.js, application.css, and all non-JS/CSS in app/assets folder are already added.

**Comment by OmTheTurtle** [comment]
> Instead of the global hash you can easily use a simple listener with the check, so something like this:
> 
> ```
> document.addEventListener("keydown", e => {
>   if (e.ctrlKey && e.key === '/') {
>       const element = document.getElementById('search-field');
>       element.focus();
>   }
> });
> ```
> 
> In this case we don't need to care the order of the listener and I think your solution won't work in some cases even if we pay attention to the order because we can be sure the listener will run in order.
> 
> By the way, do we really want to use the Ctrl+/ shortcut for the search? The simple `/` sounds better and more common as I see it.

I agree that this solution is overly complex and not really needed. In case of the shortcut, we want to avoid just using '/' as it triggers when the user writes '/' into an input field. One common shortcut that I see being used quite often is Ctrl+K we can consider using it.

**Comment by SepsiLaszlo** [comment]
Thank you all for the reviews!
I made the suggested changes, they made the code much cleaner!

I observed some interesting behivour.
`Ctlr + K` is the default browser shortcut for openning the browsers address bar.
So I had to use `preventDefault` to disable this functionality.
Also when subscribed to the `keyup` event defalut behaviour was not prevented somehow.
In order to solve this I had to subscribe to the `keydown` event.

I did not found any resource describing this.
Do you know why this behaviour happens?

**Review by balintkiraly — APPROVED** [review]

**Pull request merged**